project/CelebornBuild.scala

```diff
@@ -679,6 +679,11 @@ trait SparkClientProjects {
   
         (assembly / assemblyMergeStrategy) := {
           case m if m.toLowerCase(Locale.ROOT).endsWith("manifest.mf") => MergeStrategy.discard
+          case m if m.startsWith("META-INF/license/") => MergeStrategy.discard
+          case m if m == "META-INF/LICENSE.txt" => MergeStrategy.discard
+          case m if m == "META-INF/NOTICE.txt" => MergeStrategy.discard
+          case m if m == "LICENSE.txt" => MergeStrategy.discard
+          case m if m == "NOTICE.txt" => MergeStrategy.discard
```

Comment:
is it possible to know which jar the file comes from?

Reply:
the ideal goal here is to drop all files `**/LICENSE.*` and `**/NOTICE.*` that comes from the third party and only keep the one provided by ourselves. I don't find a elegant approach in Maven, hope we can make it happen in SBT

Reply:
Implemented a simple `ResourceTransformer` that can output `LICENSE.txt` and `NOTICE.txt` source jar.

```java
import org.apache.maven.plugins.shade.resource.ResourceTransformer;

public class PrintResourceTransformer implements ResourceTransformer {
  @Override
  public boolean canTransformResource(String s) {
    boolean hasFlag = false;
    if (s.contains("LICENSE.txt")) {
      hasFlag = true;
    } else if (s.contains("NOTICE.txt")) {
      hasFlag = true;
    }
    if (hasFlag) {
      URL resource = this.getClass().getResource("/" + s);
      System.err.printf("PrintResourceTransformer %s,%s,%s%n", s, hasFlag, resource);
    }
    return hasFlag;
  }
```


```java
PrintResourceTransformer META-INF/LICENSE.txt,true,jar:file:/Users/csy/.m2/repository/org/jdom/jdom2/2.0.6.1/jdom2-2.0.6.1.jar!/META-INF/LICENSE.txt
PrintResourceTransformer META-INF/NOTICE.txt,true,jar:file:/Users/csy/.m2/repository/commons-io/commons-io/2.6/commons-io-2.6.jar!/META-INF/NOTICE.txt
```

Reply:
unfortunately, the built-in apache license/notice transformer does not work well during my previous test

Reply:
> unfortunately, the built-in apache license/notice transformer does not work well during my previous test

`ApacheLicenseResourceTransformer` just ignores `META-INF/LICENSE`, `META-INF/LICENSE.txt`, `META-INF/LICENSE.md`.


https://github.com/apache/maven-shade-plugin/blob/master/src/main/java/org/apache/maven/plugins/shade/resource/ApacheLicenseResourceTransformer.java#L31

Reply:
@cxzl25 can we implement it in sbt? the previous discussion aims to remove maven and only keep sbt finally.

Reply:
How about using contains? I tested it locally and verified that it can be excluded.